### DISCUSSION TITLE
Routing with alerts?

### DISCUSSION THREAD

**Discussion by rhurlbatt** [anvilistas/anvil-extras#231] [Ideas]
Is there any way to use routing with forms loaded in alerts?
I often use forms loaded in to alerts to edit data while the MainForm stays loaded with other data.
These alerts can be called from modules or other forms

Is this even possible?

**Comment by s-cork** [comment]
Yes. In the docs at hash routing there's a section about this. 

The section is something like - "sometimes my route form is an alert". 

Just load the form in the alert with route=False. 

```python
alert(MyForm(route=False))
```

Although obviously this won't be part of back forward navigation.

**Comment by rhurlbatt** [comment]
Thanks Stu.

My idea was actually to have the forms added to the history stack even though they were in an alert.

For example, let's say we have `AllItemsForm` loaded in to the main form and we call an alert from `AllItemsForm` that has the `EditItemForm` embedded in to it but  we want the alert with the `EditItemForm` added to the history stack.

My initial work-around idea is to have a blank form with a `content_panel` that dynamically adds the `AllItemsForm` to the content_panel and then calls an alert with `EditItemForm` .

This way you can still add to the stack but still have the alert.

Your idea #240 might help with some of this if `@routing.route()` could be applied to things other than forms?

While I am writing this - I am not actually sure if this is really that valuable - the main reason for me initialling calling an alert was so that I didn't have to unload the page - routing makes this a whole lot easier since you don't have to manually cache the form...

**Comment by s-cork** [comment]
I can't see it being a feature that's easy to add. Mostly because when an alert is opened anvil the form that opens it might depend on the return value...

**Comment by rhurlbatt** [comment]
I would like to have a crack at doing this ... but it is going to take me some time to get up to speed with the code (and my proficiency!!) I have only just started using hash routing so I am still finding my way around it but would really like to contribute.

**Reply by s-cork** [reply]
Sure - though fair warning - the code base is dense for the part of routing that you'll end up touching.
And I think it might end up being a sizeable feature - since it's hard coded that when the url_hash changes - the content of the content panel is cleared and replaced by the new route form.

**Reply by rhurlbatt** [reply]
Ok great - sounds like a good challenge - let me wet my feet a little with first and I will see how I go
